### ISSUE TITLE
Support optional env_files

### ISSUE THREAD

**Issue by starkat99** [sagiegurari/cargo-make#582]
### Feature Description
Currently, it's an error if an `env_files` path doesn't exist. I'd like to be able to ignore that error when the file doesn't exist and continue as normal

### Describe The Solution You'd Like
Add an `optional` flag to `env_files` config that when true will ignore the file if the file doesn't exist.

### Code Sample

```toml
env_files = [ { path = "env.env", optional = true } ]
```

**Comment by sagiegurari** [comment]
how about something like this (didn't test)?

```toml
env_scripts = [
'''
#!@duckscript
${file} = set ./myfile.env
file_exists = is_file ${file}
if ${file_exists}
  text = readfile ${file}
  lines = split ${text} \n
  for line in ${lines}
    parts = split ${line} =
    count = array_length ${parts}

    if eq ${count} 2
      value = array_pop ${parts}
      key = array_pop ${parts}

      # just in case
      key = trim ${key}
      value = trim ${value}

      set_env ${key} ${value}
    end

    release ${parts}
  end
end
'''
]
```

**Comment by sagiegurari** [comment]
I'm releasing a new duckscript version that would make it even simpler:

```toml
# load env file
env_scripts = [
'''
#!@duckscript
text = readfile ./myfile.env
handle = map
map_load_properties ${handle} ${text}
set_env --handle ${handle}
'''
]
```

**Comment by sagiegurari** [comment]
closing due to lack of feedback and since there is a simple solution for users.

**Comment by skhalymon** [comment]
Hey @sagiegurari! Actually, the solution that you've provided is good, but not the best. I've faced that CI don't want to work with script, while locally it works. So `optional = true` would be nice option here.

**Comment by sagiegurari** [comment]
what donyou mean doesn't work with script?
duckscript runtime is embedded inside cargo-make.

**Comment by AlexAegis** [comment]
Hey @sagiegurari, just migrated all my projects to cargo-make, super useful, love it so far!

I was trying to do something like this and didn't find anything related to this in the readme. When I search for the word "optional" I came across another feature that had a flag like this so I just instinctively tried it with env_files too, it didn't work so I searched the github issues if this was already requested or not.

Now I see that there's a workaround but it's not optimal because it's not documented and it's very hard to discover.

**Comment by sagiegurari** [comment]
@AlexAegis its really hard to follow closed issues

you put duckscript in env_files instead of env_scripts. look at this example: https://github.com/sagiegurari/cargo-make/blob/master/examples/env.toml#L8
in addition you have [is_file ](https://github.com/sagiegurari/duckscript/blob/master/docs/sdk.md#stdfsisfile)command that you can condition to make sure file exists before loading

**Comment by caemor** [comment]
I combined the first and second (not working) workaround for a more compact version:

```toml
# env_files =  [ ".env" ]
# Workaround for optionally loading environment variables from a file
# https://github.com/sagiegurari/cargo-make/issues/582
env_scripts = [
'''
#!@duckscript
${file} = set ./myfile.env
file_exists = is_file ${file}
if ${file_exists}
    text = readfile .env
    handle = map
    map_load_properties ${handle} ${text}
    set_env --handle ${handle}
end
'''
]
```
